### DISCUSSION TITLE
Taiko Skinning

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#17119] [Ideas]
osu!taiko skinning is disparately behind all other game modes. While some may consider these unnecessary, many players consider them imperative to gameplay:
1. (#5921) This is especially apparent as notes can appear without centers rendered when viewing maps in the editor:
![osu_2022-03-06_00-08-06](https://user-images.githubusercontent.com/74262407/156914773-a8787404-507d-4dfd-a348-9a1fcc9bcb7c.png)
2. (#9858) While not a large issue, this is extremely noticeable when migrating legacy skins to lazer.
3. (#6124 and #12871) This is imperative. The LegacyComboCounter cannot be placed inside the hit target due to its animation. Additionally, the DefaultComboCounter is not center-justified in its text meaning that it cannot be centered inside the hit target.
4. (#13601) This applies to drumrolls as well; these are basic hitobjects and should be easily skinnable.

This doesn't feel like a lot, but will improve the osu!taiko onboarding process tenfold.
Highlighting these issues here in one centralized location after today's community meeting.

**Comment by bdach** [comment]
not sure what is the point of this discussion other than to mention that issues are still open need work (which is self-explanatory).

combo counter stuff is also covered under #10526. also drumrolls are skinnable as far as i know.

**Reply by Walavouchey** [reply]
> not sure what is the point of this discussion other than to mention that issues are still open need work (which is self-explanatory).

potentially to request bumping them in priority?

**Reply by zachmanthethird** [reply]
Correct. After the most recent community meeting, it seemed apparent that you believe scores are submittable to be worth ranked score and performance points.
However, the osu!taiko community strongly disagrees, and this was intended as an illustration of the skin elements that many taiko players would like to see implemented.

**Reply by peppy** [reply]
How does skinning relate in any way to that discussion? I think you might be a bit confused. Feature parity is a separate goal.

**Comment by Joppe27** [comment]
Just going to leave these missing taiko skin features here, these don't have issues assigned yet:

- ~~Kiai fire is not implemented yet. This is the `taiko-glow` element.~~ (#21459)
- ~~Animated hitobjects are not supported yet. These are the `overlay-n` elements.~~ (#21379)
- Taiko hitsounds should fall back to the legacy taiko sounds when files are missing. Not to `normal-hitnormal`, `normal-hitclap` etc. (those are osu!standard sounds).
- The combo announcer is not supported yet. These are the `comboburst-n` elements.
- osu!(standard) elements in beatmap-included skins should not be used for taiko maps.
- ~~Kiai flashing of the taiko hitobjects is not supported yet.~~ (#20819)
- `taiko-bar-right-glow` filling for widescreen is different from stable behaviour. On lazer the texture gets stretched to fit the playfield, on stable it gets stretched by a fixed factor of 3.5x anchored to the left (I think, it's kind of a mess). Lazer arguably does it better but this does change (some) skins in a small way, doesn't necessarily need to be changed imo.

**Comment by zachmanthethird** [comment]
Since posting this, several members of the community have informed me that not only are hit animations not skinnable, they are not able to be hidden, even with the UI hidden using Shift+Tab.

**Reply by peppy** [reply]
Shift+Tab is used to hide UI elements. It would obviously not hide hit animations. Please refrain from posting these comments as they are noise.

**Comment by zachmanthethird** [comment]
I believe I should make my intentions here more clear.
I know you don't like Twitch clips, but this was cut from the YT reupload, so I feel the need to link this here. https://www.twitch.tv/videos/1417003846?t=00h53m25s From 53:25 to 53:50, you've mentioned that "if you find an issue, bring it up please as a priority." This is what I've done here.
While these may seem like petty issues to you, these are make/break decisions for many osu!taiko players. I believe that these are a huge priority, though the progress on the linked issues doesn't support that thought process.
> Shift+Tab is used to hide UI elements. It would obviously not hide hit animations.

This is what stable does? I'm talking about how notes fly across and eventually leave the left side of the screen. Even if this is not disabled with Shift+Tab, this is a huge distraction to many players and needs to be disable-able somewhere.
